pkg/agent/packetcapture/capture/bpf.go

```diff
@@ -92,23 +173,64 @@ func loadIPv4HeaderOffset(skipTrue uint8) []bpf.Instruction {
 	}
 }
 
-func compareProtocolIP4(skipTrue, skipFalse uint8) bpf.Instruction {
-	return bpf.JumpIf{Cond: bpf.JumpEqual, Val: etherTypeIPv4, SkipTrue: skipTrue, SkipFalse: skipFalse}
+func compareProtocolIP(etherType uint32, skipTrue, skipFalse uint8) bpf.Instruction {
+	return bpf.JumpIf{Cond: bpf.JumpEqual, Val: etherType, SkipTrue: skipTrue, SkipFalse: skipFalse}
 }
 
 func compareProtocol(protocol uint32, skipTrue, skipFalse uint8) bpf.Instruction {
 	return bpf.JumpIf{Cond: bpf.JumpEqual, Val: protocol, SkipTrue: skipTrue, SkipFalse: skipFalse}
 }
 
-func calculateSkipFalse(srcIP, dstIP net.IP, transport *transportFilters) uint8 {
+// getAddressChunk abstracts the process of extracting a 4-byte chunk from an IP address,
+// handling the structural differences between IPv4 (one chunk) and IPv6 (four chunks).
+func (h *ipFamilyHandler) getAddressChunk(ip net.IP, chunkIndex int) uint32 {
+	if h.etherType == etherTypeIPv4 {
+		return binary.BigEndian.Uint32(ip[len(ip)-4:])
+	}
+	return binary.BigEndian.Uint32(ip[chunkIndex*4 : (chunkIndex+1)*4])
+}
+
+// calculateSkipOffset determines the correct 'SkipFalse' jump offset for an IP address chunk.
+func (h *ipFamilyHandler) calculateSkipOffset(chunkIndex int, skipFalse, skipToEnd uint8, needsCheck bool) uint8 {
+	if needsCheck {
+		if h.etherType == etherTypeIPv4 {
+			return skipFalse
+		}
+		return skipFalse - uint8((chunkIndex+1)*2)
```

Comment:
can you add more description for this function? we have too many things seems totally clueless at first in this file..